From e19f4b1a5ccd682b2d67171f29a037e0b754085b Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Sat, 25 Oct 2014 12:20:36 +0100
Subject: Substitute bindir and libexecdir in Makefile.am

Rather than substituting the values of libexecdir and bindir during
configure, which might include some unexpanded shell variables, expand
and substitute them as targets in Makefile.am instead.

https://bugzilla.gnome.org/show_bug.cgi?id=726095

diff --git a/Makefile.am b/Makefile.am
index 469e3a4..771cb74 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -367,10 +367,17 @@ desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 appstream_in_files = data/org.gnome.Cheese.appdata.xml.in
 appstream_XML = $(appstream_in_files:.appdata.xml.in=.appdata.xml)
 
+data/org.gnome.Cheese.service: data/org.gnome.Cheese.service.in
+	$(AM_V_GEN)$(SED) -e "s|[@]bindir[@]|$(bindir)|" $< > $@
+
+data/org.gnome.Camera.service: data/org.gnome.Camera.service.in
+	$(AM_V_GEN)$(SED) -e "s|[@]libexecdir[@]|$(libexecdir)|" $< > $@
+
 servicedir = $(datadir)/dbus-1/services
-service_DATA = \
-	data/org.gnome.Camera.service \
-	data/org.gnome.Cheese.service
+service_in_files = \
+	data/org.gnome.Camera.service.in \
+	data/org.gnome.Cheese.service.in
+service_DATA = $(service_in_files:.service.in=.service)
 
 @GSETTINGS_RULES@
 gsettings_SCHEMAS = data/org.gnome.Cheese.gschema.xml
@@ -534,11 +541,12 @@ dist-hook: git-changelog-hook
 dist_noinst_DATA = \
 	ChangeLog.pre-git \
 	COPYING.GPL3 \
+	$(appstream_in_files) \
 	$(desktop_in_files) \
 	$(gsettings_SCHEMAS) \
 	$(noinst_gnome_camera_service_headers) \
 	$(noinst_resource_files) \
-	$(appstream_in_files) \
+	$(service_in_files) \
 	data/org.gnome.Cheese.gresource.xml \
 	data/org.gnome.Camera.xml \
 	build-aux/test-driver \
@@ -566,6 +574,7 @@ CLEANFILES = \
 	$(pkgconfig_DATA) \
 	$(enum_data) \
 	$(gir_DATA) \
+	$(service_DATA) \
 	$(typelib_DATA) \
 	$(nodist_gnome_camera_service_headers) \
 	$(nodist_gnome_camera_service_SOURCES) \
diff --git a/configure.ac b/configure.ac
index b32be22..13843f7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -215,9 +215,7 @@ docs/reference/Makefile
 docs/reference/version.xml
 data/cheese.pc
 data/cheese-gtk.pc
-data/org.gnome.Camera.service
 data/org.gnome.Cheese.desktop.in
-data/org.gnome.Cheese.service
 help/Makefile
 po/Makefile.in
 ])
-- 
cgit v0.10.1

